ape_infura/providers.py

```diff
@@ -36,49 +36,75 @@ def __post_init__(self):
             raise MissingProjectKeyError()
 
         self._web3 = Web3(HTTPProvider(f"https://{self.network.name}.infura.io/v3/{key}"))
+        if self._web3.eth.chain_id in (4, 5, 42):
```

Comment:
Was thinking about eventually making a chain ID enum based off of https://chainlist.org/

Reply:
Good idea! A simple enum with just the common chain IDs is probably good enough for now. I'll create a ticket for that.

But we could do something like dynamically generate the enum class using this request: https://chainid.network/chains.json, that'd be wild.

Reply:
https://github.com/ApeWorX/ape/issues/254